### PR TITLE
Add MTE-2571 - zoom actions validation test

### PR DESCRIPTION
**Pull request by dragosb01**
## :scroll: Tickets
https://mozilla-hub.atlassian.net/browse/MTE-2571

## :bulb: Description
Partially automated pan/zoom actions test case
https://testrail.stage.mozaws.net/index.php?/cases/view/2306947

Screenshot 2024-03-28 at 18 07 03

### CODE REVIEW

**Review by isabelrios — APPROVED** [review]

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift [hunk 1]
```diff
@@ -357,6 +357,13 @@ class BaseTestCase: XCTestCase {
             nrOfSwipes += 1
         }
     }
+
+    func panScreen() {
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift:361]: I think this depends on the simulator used, let's add a comment with the one used

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        navigator.goto(TabTray)
+        if !app.buttons[AccessibilityIdentifiers.TabTray.newTabButton].exists {
+            app.buttons[AccessibilityIdentifiers.Toolbar.tabsButton].tap()
+        }
+        navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
+        validateZoomActions()
+    }
+
+    func validateZoomActions() {
+        navigator.openURL("www.mozilla.org")
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift:40]: can we use a local served site instead?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        XCUIDevice.shared.orientation = UIDeviceOrientation.landscapeLeft
+        zoomOutLandscape()
+        zoomInLandscape()
+        zoomInButton.tap()
+        mozWaitForElementToExist(app.staticTexts[AccessibilityIdentifiers.ZoomPageBar.zoomPageZoomLevelLabel])
+        zoomLevel = app.staticTexts[AccessibilityIdentifiers.ZoomPageBar.zoomPageZoomLevelLabel]
+        XCTAssertEqual(zoomLevel.label, "Current Zoom Level: 100%")
+    }
+
+    func zoomIn() {
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift:69]: Wondering if we should add these methods in the BaseTestCase instead...

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift (hunk 2 above)
Comment (dragosb01) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/ZoomingTests.swift]: updated

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift (hunk 1 above)
Comment (dragosb01) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift:362]: updated

**Pull request merged**